doc/devel/cluster/architectures.rst

```diff
@@ -0,0 +1,249 @@
+   event to all workers. This is unnecessary with non-Broker backends. It
+   still works, but is less efficient than sending remote events directly to
+   the destination topic.
+
+   References
+
+   * https://github.com/zeek/zeek/issues/3917
+   * https://github.com/zeek/zeek/discussions/3649
+
+
+
+
+Single Node Examples
+====================
+
+A Zeek cluster can be deployed on a single hardware/virtual system.
+This looks as follows when the system has a single monitoring NIC.
+
+.. figure:: /images/cluster/single-system-one-nic.svg
+
+This example depicts load-balancing of all network traffic across 8 workers.
+
+When monitoring two NICs on the same system, the cluster looks
+as follows, assuming four workers are sufficient per NIC.
+
+.. figure:: /images/cluster/single-system-two-nics.svg
+
+
+Note that while these diagrams depict just a few workers per NIC, high-performance
+environments may require 100 or more Zeek workers and - depending on the loaded
+scripts - up to 10s of proxies. Such environments generally require experiments,
+monitoring and performance tuning to establish the appropriate number of workers,
+loggers and proxies for a given hardware configuration and system.
+
+.. note::
+
+   Zeek loggers are an artificial bottleneck in a Zeek cluster. Particular if all
+   Zeek logs are forwarded to messaging queue like Kafka or NATS, it is almost
+   certainly more efficient to have workers send log to the message queue
+   directly instead of funneling them through an extra Zeek process.
+   The tagline here is that Kafka or NATS are likely better Zeek loggers.
+
+   As of Zeek 8.0, there is no ready-to-use recipe for running a logger-less
+   cluster. Contributions in this area are certainly welcome.
+
+   For the classic central file-based logging. a single Zeek logger is useful
+   as the aggregation point.
+
+
+Multi Node Examples
+===================
+
+Zeek was originally developed during a time in which multi-core CPUs weren't
+widely available and scaling and flow-balancing was actually done on the level
+of a full hardware system.
+This can still be useful today for deploying multiple smaller worker systems
+with 10G NICs compared to a single large system with hundreds of CPUs and a
+powerful 100G NIC. Such an architecture requires an external packet broker
+that ensures consistent flow-balancing across the individual Zeek systems,
+essentially introducing a separate flow-balancing layer.
+
+Shared Publish Subscribe Layer
+------------------------------
+
+The following diagram depicts a deployment with three individual hardware
+systems with the system to the left running a single logger and manager process.
+Each system runs a single proxy process (proxy-1, proxy-2 and proxy-3) and
+four worker processes.
+All Zeek processes can send remote events to each other through the shared
+publish subscribe layer.
+
+
+.. figure:: /images/cluster/three-systems-one-nic.svg
```

Comment:
these more zoomed out diagrams are impossible for me to read, but each separate host could probably just be a box rather than all of that detail?